src/Stack/Build/Execute.hs

```diff
@@ -801,6 +825,13 @@ withSingleContext runInBase ActionContext {..} ExecuteEnv {..} task@Task {..} md
             logPath <- buildLogPath package msuffix
             ensureDir (parent logPath)
             let fp = toFilePath logPath
+
+            -- We only want to dump logs for local non-dependency packages
+            case taskType of
+                TTLocal lp | lpWanted lp ->
```

Comment:
I'm not convinced that using `lpWanted` is the right thing here, we may instead want to check if this is listed as an `extra-dep` or not. Feedback welcome.